tuf/api/keys.py

```diff
@@ -0,0 +1,169 @@
+        self.private = private
+        self.public = public
+
+class Key:
+
+    def __init__(self, path: str, obj: Any):
+        self.path = path
+        self.obj = obj
+
+class Keypair:
+
+    def __init__(self, private: Key, public: Key):
+        self.private = private
+        self.public = public
+
+Keypairs = List[Keypair]
+
+class Keyring:
+
+    def __init__(self, threshold: Threshold, keypairs: Keypairs):
+        if len(keypairs) >= threshold.m:
+            logging.warning(f'{len(keypairs)} >= {threshold.m}')
+        if len(keypairs) <= threshold.n
+            logging.warning(f'{len(keypairs)} <= {threshold.n}')
+        self.threshold = threshold
+        self.keypairs = keypairs
+
+# Useful for securesytemslib.
+KeyDict = Dict[str, Any]
+
+# Utility functions.
+
+def get_new_private_keypath(keystore_dir: str, rolename: str, i : int = 1) -> str:
+    return os.path.join(keystore_dir, f'{rolename}_ed25519_key_{i}')
+
+def get_public_keypath(private_keypath: str) -> str:
+    # this is the tuf filename convention at the time of writing.
+    return f'{private_keypath}.pub'
+
+def get_private_keys_from_keyring(keyring: Keyring) -> KeyDict:
+    privkeys = {}
+
+    for keypair in keyring.keypairs:
+        privkey = keypair.private.obj
+        keyid = privkey['keyid']
+        assert keyid not in privkeys
+        privkeys[keyid] = privkey
+
+    return privkeys
+
+def get_public_keys_from_keyring(keyring: Keyring) -> KeyDict:
+    pubkeys = {}
+
+    for keypair in keyring.keypairs:
+        pubkey = keypair.public.obj
+        keyid = pubkey['keyid']
+        assert keyid not in pubkeys
+        pubkeys[keyid] = pubkey
+
+    return pubkeys
+
+def write_keypair(keystore_dir: str, rolename: str, i: int = 1, n: int = 1, passphrase: Optional[str] = None) -> Keypath:
+    private_keypath = get_new_private_keypath(keystore_dir, rolename, i)
+    assert not os.path.isfile(private_keypath)
+    public_keypath = get_public_keypath(private_keypath)
+    assert not os.path.isfile(public_keypath)
+
+    # Make the keystore directory, WR-only by self, if not already there.
+    os.makedirs(keystore_dir, mode=0o700, exist_ok=True)
+
+    # FIXME: do not assume Ed25519
+    generate_and_write_ed25519_keypair(private_keypath, password=passphrase)
+
+    return Keypath(private_keypath, public_keypath)
+
+def read_keypair(keypath: Keypath, passphrase: Optional[str] = None) -> Keypair:
+    private_keypath = keypath.private
+    private_key_obj = import_ed25519_privatekey_from_file(keypath.private, password=passphrase)
+    private_key = Key(private_keypath, private_key_obj)
+
+    # and its corresponding public key.
+    public_keypath = keypath.public
+    public_key_obj = import_ed25519_publickey_from_file(keypath.public)
+    public_key = Key(public_keypath, public_key_obj)
+
+    return Keypair(private_key, public_key)
+
+def rename_keys_to_match_keyid(keystore_dir: str, keypair: Keypair) -> None:
```

Comment:
This seems like a crutch that we only need because we don't write the keys to the right location in the first place. Maybe we can rather fix this at the source of that problem (in sslib) instead of working around here?